packages/shared/src/store/storage/upload.ts

```diff
@@ -121,31 +126,70 @@ export const useFileStore = create<FileStore>((set, get) => ({
       draft.files = { ...draft.files, ...newFiles };
     });
 
-    fileList.forEach((f) => upload(uploader, f, config, imageSizer));
+    fileList.forEach((f) =>
+      upload(uploader, f, config, imageSizer, undefined, nativeUploader)
+    );
   },
-  upload: async (uploader, upload, config, imageSizer, compressor) => {
+  upload: async (
+    uploader,
+    upload,
+    config,
+    imageSizer,
+    compressor,
+    nativeUploader
+  ) => {
     const { client, updateStatus, updateFile } = get();
 
     const { key, file } = upload;
     updateStatus(uploader, key, 'loading');
+    const isHostedUpload =
+      config.service === 'presigned-url' && config.presignedUrl;
+    const isCustomUpload = client !== null;
 
     // TODO: implement image compression for react-native
-    // const compressionOptions = {
-    // maxSizeMB: 1,
-    // useWebWorker: true,
-    // };
-
-    // if compression fails for some reason, we'll just use the original file.
-    // let compressedFile: File = file;
-
-    // try {
-    // // TODO: fix upload compression for Safari and iOS webview
-    // if (isImageFile(file)) {
-    // compressedFile = await imageCompression(file, compressionOptions);
-    // }
-    // } catch (error) {
-    // console.log({ error });
-    // }
+
+    // Native app uses its own uploader which gets passed in
+    if (nativeUploader) {
```

Comment:
Does this still handle resizing + compressing the image? I think those are important to get in as massive imaages cause all kinds of pain, and photo roll images are pretty massive by default.

Reply:
That piece should be unchanged. We resize the image in `useImageUpload` before it ever gets passed to the uploader in file store.

Reply:
Ahh, ok, I see where I tripped up. Not related to your PR, but while you're in there, would you mind renaming `imageSizer` to `getImageSize`? I think that'd be clearer.